ixwebsocket/IXHttpClient.cpp

```diff
@@ -260,7 +263,7 @@ namespace ix
         }
 
```

Comment:
For the transfer timeout, _stop will not be read which sounds incorrect.

Also can you edit the doc to add description for your new cancel argument ?

Reply:
I think we just need to replicate the logic you added with your lambda down here.

Also I think it would be good to have a unittest for that new code.

Reply:
I added a unit test + a special error (`HttpErrorCode::Cancelled`) to indicate that the request was cancelled and I also updated the docs to show how cancellation is used.

According to my understanding of C++ semantics it shouldn't be necessary to replicate the lambda further down. It has a reference to the `cancelled` variable which we update later. When `cancelled` is reassigned that change is reflected in the lambda (because it's a reference and not a copy).

Reply:
Thanks, I'll take a look.